components/director/pkg/oauth/config.go

```diff
@@ -14,3 +23,51 @@ type Config struct {
 	TokenRequestTimeout   time.Duration `envconfig:"APP_OAUTH_TOKEN_REQUEST_TIMEOUT"`
 	SkipSSLValidation     bool          `envconfig:"APP_OAUTH_SKIP_SSL_VALIDATION"`
 }
+
+type X509Config struct {
+	Cert string `envconfig:"APP_OAUTH_X509_CERT,optional"`
+	Key  string `envconfig:"APP_OAUTH_X509_KEY,optional"`
+}
+
+func (c *X509Config) ParseCertificate() (*tls.Certificate, error) {
```

Comment:
Almost the same method is used in the certificate loader - https://github.com/kyma-incubator/compass/blob/main/components/director/pkg/certloader/loader.go#L148
Maybe we should extract it somewhere where we can use it from both places.
We have a`cert` package which holds some utils, used by `connector` and `director`, maybe we can put it there?

Reply:
done